python/pyarrow/tests/test_compute.py

```diff
@@ -974,6 +974,51 @@ def test_take_null_type():
     assert len(table.take(indices).column(0)) == 4
 
 
+@pytest.mark.parametrize(('ty', 'values'), all_array_types)
+def test_dropnull(ty, values):
+    arr = pa.array(values, type=ty)
+    result = arr.dropnull()
+    result.validate()
+    indices = [i for i in range(len(arr)) if arr[i].is_valid]
+    expected = arr.take(pa.array(indices))
+    assert result.equals(expected)
+
+
+def test_dropnull_chunked_array():
+    arr = pa.chunked_array([["a", None], ["c", "d", None]])
+    expected_drop = pa.chunked_array([["a"], ["c", "d"]])
+    result = arr.dropnull()
+    assert result.equals(expected_drop)
+
+
+def test_dropnull_record_batch():
+    batch = pa.record_batch(
+        [pa.array(["a", None, "c", "d", None])], names=["a'"])
```

Comment:
Can you make some tests for record batches with multiple columns? (I think this will fail if they have different numbers of null elements).